libsol/system_instruction.c

```diff
@@ -59,7 +40,7 @@ int parse_system_transfer_instructions(Parser* parser, MessageHeader* header, Sy
     Pubkey system_program_id = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
     BAIL_IF(memcmp(program_id, &system_program_id, PUBKEY_SIZE));
 
-    BAIL_IF(parse_system_transfer_instruction(&instruction, header->pubkeys, header->pubkeys_length, info));
+    BAIL_IF(parse_system_transfer_instruction(&instruction, header->pubkeys, header->pubkeys_header.pubkeys_length, info));
```

Comment:
@t-nelson, for the curious, here was the bug: I moved `pubkeys_length` to a different struct and forgot to delete the original, leaving an uninitialized one and one spot where the code referenced it.